### PR TITLE
532 bri3 transactions read api

### PR DESCRIPTION
**Pull request by ognjenkurtic**
# Description

Adds read by id functionality to the transaction API from PR #533 .Should be reviewed after that one is merged.
Didnt add get all functionality as we still do not have from and to fields so it does not make much sense to read all transactions from everyone. 
Tests are coming in a final, separate PR.

## Related Issue

#532 

## How Has This Been Tested

Manual testing through the provided Postman collection

## Types of changes

- [] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      transactionModel.workstepInstanceId,
+      null, // TODO: transactionModel.from once BpiAccount in the prisma schema,
+      null, // TODO: transactionModel.to once BpiAccount in the prisma schema,
+      transactionModel.payload,
+      transactionModel.signature,
+      transactionModel.status,
+    );
+  }
+
+  async getAllTransactions(): Promise<Transaction[]> {
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts:30]: see my comments from the related RP #533

File: examples/bri-3/src/bri/transactions/agents/transactions.agent.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    if (transactionToUpdate.status != TransactionStatus.Initialized) {
+      throw new BadRequestException(UPDATE_WRONG_STATUS_ERR_MESSAGE);
+    }
+
+    return transactionToUpdate;
+  }
+
+  public updateTransaction(
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/transactions/agents/transactions.agent.ts:63]: @ognjenkurtic see my comment in #533 as to what elements in a transaction  could be updated

File: examples/bri-3/src/bri/transactions/api/dtos/request/updateTransaction.dto.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { IsNotEmpty } from 'class-validator';
+
+export class UpdateTransactionDto {
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/transactions/api/dtos/request/updateTransaction.dto.ts:3]: related to my comment above about what elements in a transaction could be changed @ognjenkurtic  

would require changes accordingly

File: examples/bri-3/src/bri/transactions/api/transactions.controller.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        requestDto.fromAccountId,
+        requestDto.toAccountId,
+        requestDto.payload,
+        requestDto.signature,
+      ),
+    );
+  }
+
+  @Put('/:id')
+  async updateTransaction(
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/transactions/api/transactions.controller.ts:47]: @ognjenkurtic see my related comments above

File: examples/bri-3/src/bri/transactions/models/transaction.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  public get signature(): string {
+    return this._signature;
+  }
+
+  public get status(): TransactionStatus {
+    return this._status;
+  }
+
+  public updatePayload(payload: string, signature: string): void {
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/transactions/models/transaction.ts:73]: seem y comments on same function in #533

**Review by Therecanbeonlyone1969 — CHANGES_REQUESTED** [review]
See my comments, all related to comments in #533

File: examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts (hunk 1 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts]: resolved

File: examples/bri-3/src/bri/transactions/agents/transactions.agent.ts (hunk 2 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/transactions/agents/transactions.agent.ts:64]: resolved

File: examples/bri-3/src/bri/transactions/api/dtos/request/updateTransaction.dto.ts (hunk 3 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/transactions/api/dtos/request/updateTransaction.dto.ts:3]: resolved

File: examples/bri-3/src/bri/transactions/api/transactions.controller.ts (hunk 4 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/transactions/api/transactions.controller.ts:47]: resolved

File: examples/bri-3/src/bri/transactions/models/transaction.ts (hunk 5 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/transactions/models/transaction.ts:73]: resolved

**Review by skosito — APPROVED** [review]

**Review by Therecanbeonlyone1969 — APPROVED** [review]
Lgtm now

**Review by Manik-Jain — APPROVED** [review]

**Pull request merged**